src/Microsoft.AspNetCore.Rewrite/Internal/Shared/Pattern.cs

```diff
@@ -5,10 +5,12 @@
 using System.Text;
 using Microsoft.AspNetCore.Http;
 
-namespace Microsoft.AspNetCore.Rewrite.Internal.UrlRewrite
+namespace Microsoft.AspNetCore.Rewrite.Internal.Shared
```

Comment:
Shared items can stay in the parent Internal namespace, or put them in a sub-namespace that describes their function.